File: API/ICCMS-API/Models/Document.cs
using Google.Cloud.Firestore;

namespace ICCMS_API.Models
{
    [FirestoreData]
    public class Document
    {
        [FirestoreProperty("documentId")]
        public string DocumentId { get; set; } = string.Empty;

        [FirestoreProperty("projectId")]
        public string ProjectId { get; set; } = string.Empty;

        [FirestoreProperty("fileName")]
        public string FileName { get; set; } = string.Empty;

        [FirestoreProperty("status")]
        public string Status { get; set; } = string.Empty;

        [FirestoreProperty("fileType")]
        public string FileType { get; set; } = string.Empty;

        [FirestoreProperty("fileSize")]
        public long FileSize { get; set; }

        [FirestoreProperty("fileUrl")]
        public string FileUrl { get; set; } = string.Empty;

        [FirestoreProperty("uploadedBy")]
        public string UploadedBy { get; set; } = string.Empty;

        [FirestoreProperty("uploadedAt")]
        public DateTime UploadedAt { get; set; }

        [FirestoreProperty("description")]
        public string Description { get; set; } = string.Empty;
    }
}

File: Mobile/app/src/main/java/com/example/iccms_mobile/data/models/Document.kt
package com.example.iccms_mobile.data.models

import com.google.gson.annotations.SerializedName

data class Document(
    @SerializedName("documentId")
    val documentId: String = "",

    @SerializedName("projectId")
    val projectId: String = "",

    @SerializedName("fileName")
    val fileName: String = "",

    @SerializedName("status")
    val status: String = "active", // default matches API

    @SerializedName("fileType")
    val fileType: String = "",

    @SerializedName("fileSize")
    val fileSize: Long = 0L,

    @SerializedName("fileUrl")
    val fileUrl: String = "",

    @SerializedName("uploadedBy")
    val uploadedBy: String = "",

    @SerializedName("uploadedAt")
    val uploadedAt: String = "",

    @SerializedName("description")
    val description: String = ""
)
